File: common-lib/src/main/java/domain/process/ProcessNode.java
package domain.process;

import lombok.Data;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.Table;

import java.util.List;
import java.util.Map;
import java.util.UUID;

@Data
@Table("process_nodes")
public class ProcessNode {
    @PrimaryKey
    private UUID id;

    private UUID localId;
    private ProcessNode previousNode;
    private List<ProcessNode> nextNodes;
    private NodeType nodeType;
    private String source;
    private String sourceActionId;
    private Map<String, String> actionConfiguration;
}

File: common-lib/src/main/kotlin/domain/process/ProcessNode.kt
package domain.process

import java.util.UUID

data class ProcessNode(
    private val id: String,
    private val localId: UUID,
    private val previousNode: ProcessNode?,
    private val nextNodes: List<ProcessNode>,
    private val nodeType: NodeType,
    private val source: String,
    private val sourceActionId: String,
    private val actionConfiguration: Map<String, String>
)
